nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/processor/StandardValidationContextFactory.java

```diff
@@ -20,25 +20,53 @@
 import org.apache.nifi.components.ValidationContext;
 import org.apache.nifi.controller.PropertyConfiguration;
 import org.apache.nifi.controller.ValidationContextFactory;
+import org.apache.nifi.controller.flowanalysis.FlowAnalyzer;
 import org.apache.nifi.controller.service.ControllerServiceProvider;
 import org.apache.nifi.parameter.ParameterContext;
 import org.apache.nifi.registry.VariableRegistry;
+import org.apache.nifi.validation.RuleViolationsManager;
 
 import java.util.Map;
 
 public class StandardValidationContextFactory implements ValidationContextFactory {
 
     private final ControllerServiceProvider serviceProvider;
     private final VariableRegistry variableRegistry;
+    private final RuleViolationsManager ruleViolationsManager;
+    private final FlowAnalyzer flowAnalyzer;
 
-    public StandardValidationContextFactory(final ControllerServiceProvider serviceProvider, final VariableRegistry variableRegistry) {
+    public StandardValidationContextFactory(
+        final ControllerServiceProvider serviceProvider,
+        final VariableRegistry variableRegistry
+    ) {
+        this(serviceProvider, variableRegistry, null, null);
+    }
+
+    public StandardValidationContextFactory(
+        final ControllerServiceProvider serviceProvider,
+        final VariableRegistry variableRegistry,
+        final RuleViolationsManager ruleViolationsManager,
+        final FlowAnalyzer flowAnalyzer
```

Comment:
Should these properties be added to the ValidationContextFactory? Property Validation and Flow Analysis are different operations, so combining them does not seem like the right approach.

Reply:
I think it makes sense if we don't consider the concept of Validation as validation of the properties of a component but the validation of the component itself.